main.go

```diff
@@ -115,9 +115,11 @@ func mainLoop(oracleInstance *oracle.Oracle, onchain *oracle.Onchain, cfg *confi
 		log.Info("Latest known root by oracle does not match the latest onchain. ",
 			latestOnchainRoot, " ", latestKnownRoot)
 		log.Info("The state wont be updated until the same root is reached")
+	} else if cfg.DryRun {
```

Comment:
mmm not sure about this behaviour. If DryRun is enabled, `syncedWithOnchainRoot` wont be updated since it's one or the other. I mean it will enter in else if cfg.DryRun { but wont enter in the else.

Right now `syncedWithOnchainRoot` is only used in !DryRun (not in dry run) but just in case.